pyinfra/api/host.py

```diff
@@ -245,6 +247,30 @@ def deploy(self, name: str, kwargs, data, in_deploy: bool = True):
             old_deploy_data,
         )
 
+    def get_temp_filename(self, hash_key: Optional[str] = None, hash_filename: bool = True):
+        """
+        Generate a temporary filename for this deploy.
+        """
+
+        temp_directory = self.state.config.TEMP_DIR
```

Comment:
Config should be host-scoped once state initalized? https://github.com/Fizzadar/pyinfra/issues/806

Reply:
Historical context: https://github.com/Fizzadar/pyinfra/commit/9fe1984b968a17552092ef6dc1482724a4125404